### DISCUSSION TITLE
Large Area Map splitting and stitching

### DISCUSSION THREAD

**Discussion by huseyinhzl** [AzdiarGazder/mtexTools#6] [Q&A] [answered]
Hi @AzdiarGazder,

Thank you so much for your Tools. I am starting the discussion with my questions. I am using Jeol 7100F with Oxford Nordlys EBSD detector, and collecting the large area map data with Aztec software. Since the collected LAM consists of at least 2.5 million data points, I am looking to use the split function that you have created. I am wondering:

1. After data collection, Aztec automatically align each tile to make a large area. Is the split function still usable in these maps? Could be a silly question, but I just have concerns about those white areas. If I calculate grains, may those areas be filled with imaginary boundaries?

![Phase Plot](https://github.com/AzdiarGazder/mtexTools/assets/120036622/e6b3bfdf-3feb-4e1e-8fe7-be6f6d7805c4)
**Figure 1 - Phase plot showing auto-aligned tiles**

2. How can I run the parent grain reconstruction for each split map? I can save and run them one by one, but this could be really time-consuming, so I am wondering if there is a way to run split and parent grain reconstruction in just one script (for loop maybe). I have not used coding till my PhD studies, so I am really unskilled person.

3. After doing all the grain reconstruction, how can I use the stitching method? Will the stitching give exactly the same map with parent grains?

I hope I am clear in what I am asking. If you need I can share as-mapped EBSD data.

Thank you

**Comment by AzdiarGazder** [comment, marked as answer]
Hi @huseyinhzl,

Thank you for itemising your issues in the order of their appearance, and for labeling your figures.
It makes things a lot clearer to a third person like me who is not involved with your work.

And you're very welcome to the functions in mtexTools and ORTools.
As acknowledgement, please cite the use of these function libraries in the references section of your publications.

Now, to answer your questions:

1. Within OI Aztec, your large area map (LAM) was created by stitching after accounting for errors in the movement of the SEM sample stage. That is why you have white areas around the edges of the stitched LAM map. So, before you start working on such a map, I suggest you crop a rectangular inner section (without the outer white regions) out of your OI Atzec stitched map. You can do this in a couple of ways:
 - You can crop your LAM within OI Aztec Crystal and then export the cropped *.ctf file.
 - You can  use the [**crop function**](https://github.com/AzdiarGazder/mtexTools/tree/main/crop) in mtexTools and save the cropped map as a *.ctf file.

NOTE:  In almost all demo functions in mtexTools there are two sections. 
One section is for maps using a hexagonal grid convention (e.g. -for users with the OIM Amatek system and *.ang files), and 
one section is for maps with a square grid convention (e.g. -for users like yourself with the OI system and *.ctf files).
Choose the square grid section as that applies to your OI maps.


2.  At this step, I am assuming that you are now working with your cropped LAM. Now use the [**split function**](https://github.com/AzdiarGazder/mtexTools/tree/main/split) to sub-divide your LAM for parent grain reconstruction by creating the **minimum number of sub-maps**. Choose the number of sub-maps such that your computer is able to undertake the parent grain reconstruction of the sub-map data without running into any out-of-memory errors. As a general rule of thumb, a sub-map with ~1M points should work (albeit slowly) without a problem on an ordinary desktop/laptop. Another important point to remember, the more contiguous a map (i.e. - the larger it is), the fewer the number of introduced errors during parent grain reconstruction. So there are two major benefits to minimising the number of sub-maps.


3. I do not advise using a loop to undertake parent grain reconstruction of the sub-maps. Each sub-map is subtly different in terms of its data and thus, each would (maybe) need a little tweaking of their numbers and values. Furthermore, given your computer resources, I advise you to undertake parent grain reconstruction of each sub-map separately. It may be a bit time consuming, but given your inexperience with scripting, it is also the safest and cleanest option as it avoids any unnecessary errors.  


4. At this step, I am assuming you have completed parent grain reconstruction of each sub-map. Now use the [**stitch function**](https://github.com/AzdiarGazder/mtexTools/tree/main/stitch) to stitch your sub-maps together. Since you know the amount of overlap you used and the location of each sub-map when splitting them (in point 2), you know the corresponding number of pixels of overlap needed and where to place each sub-map during stitching. While stitching **should (in theory)** return you the cropped LAM, there may be some issues caused by the reconstruction of the parent grains. That is where a little manual tweaking of sub-map positions during stitching may be needed.


5. A last bit of advice. I suggest you crop your LAM in OI Aztec Crystal. Following cropping, remove wild spikes and fill-in your map in OI Aztec Crystal while avoiding voids. After filling-in is complete, export your map as a *.ctf file for further work in MATLAB/ MTEX/ ORTools/ mtexTools. The reason I am suggesting this workflow is because it will save you processing time (and resources) during parent grain reconstruction, and reduce errors even further.

NOTE: Full cleaning of your maps is not possible due to poor map quality (i.e. - fine microstructure but coarse step size used to acquire maps). [**Please refer to this discussion thread for details.**](https://github.com/ORTools4MTEX/ORTools/discussions/51#discussioncomment-6984916)

[**For tutorials on how to use OI Aztec Crystal, please have a look at the video resources available here.**](https://www.ebsd.com/oi-ebsd-training-hub/data-processing-training-exercises) 



Hope this helps.
**If the above explanations have resolved your issues, please mark this message as your answer.**

Warm regards,
Azdi

**Reply by huseyinhzl** [reply]
Hi @AzdiarGazder,

Thank you so much for your suggestions. I will follow your suggestions.
